src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/CertificateRegistry.java

```diff
@@ -5,155 +5,81 @@
                     return size() > RegistryConfig.REGISTRY_CACHE_SIZE;
                 }
             });
 
-    private final IotAuthClient iotAuthClient;
-
     /**
-     * Constructor.
-     *
-     * @param iotAuthClient IoT Auth Client
+     * Retrieve certificate by certificate pem.
+     * @param certificatePem cert pem
+     * @return certificate object
      */
-    @Inject
-    public CertificateRegistry(IotAuthClient iotAuthClient) {
-        this.iotAuthClient = iotAuthClient;
-    }
-
-    /**
-     * Returns whether the provided certificate is valid and active.
-     * Returns valid locally registered result when IoT Core cannot be reached.
-     *
-     * @param certificatePem Certificate PEM
-     * @return true if the certificate is valid and active.
-     * @throws CloudServiceInteractionException if the certificate cannot be validated
-     */
-    public boolean isCertificateValid(String certificatePem) {
-        try {
-            Optional<String> certId = fetchActiveCertificateId(certificatePem);
-            updateRegistryForCertificate(certificatePem, certId);
-            return certId.isPresent();
-        } catch (CloudServiceInteractionException e) {
-            return getAssociatedCertificateId(certificatePem)
-                    .map(certId -> true)
-                    .orElseThrow(() -> e);
-        }
+    public Certificate getCertificateByPem(String certificatePem) {
+        return getCertificateById(getCertificateId(certificatePem));
     }
 
     /**
-     * Get IoT Certificate ID for given certificate pem.
-     * Active IoT Certificate Ids are cached locally to avoid multiple cloud requests.
-     *
-     * @param certificatePem Certificate PEM
-     * @return IoT Certificate ID or empty Optional if certificate is inactive/invalid
-     * @throws IllegalArgumentException for empty certificate PEM
-     * @throws CloudServiceInteractionException if IoT certificate Id cannot be fetched
+     * Retrieve certificate by certificate id.
+     * @param certificateId cert id
+     * @return certificate object
      */
-    public Optional<String> getIotCertificateIdForPem(String certificatePem) {
-        if (Utils.isEmpty(certificatePem)) {
-            throw new IllegalArgumentException("Certificate PEM is empty");
-        }
-        try {
-            Optional<String> certId = getAssociatedCertificateId(certificatePem).map(Optional::of)
-                    .orElseGet(() -> fetchActiveCertificateId(certificatePem));
-            updateRegistryForCertificate(certificatePem, certId);
-            return certId;
-        } catch (CloudServiceInteractionException e) {
-            return getAssociatedCertificateId(certificatePem).map(Optional::of)
-                    .orElseThrow(() -> e);
+    public Certificate getCertificateById(String certificateId) {
+        Instant lastValid = registry.get(certificateId);
+        if (lastValid != null) {
```

Comment:
I don't know about this. This is not actually doing a lookup of a certificate by its id on the registry it is more of creating a certificate and having some business logic (It knows that is last valid is not null the status should be "ACTIVE"). Ideally that logic is part of the certificate itself and we store certificates in the repository. Maybe what should be stored is a `Map<String, Certificate>` and then this would

Reply:
Don't worry about the storage bits. Same comment as thing registry. This is all changing anyway.
But fair point about keeping business logic out of this. Agree there